### PR TITLE
Add some str annotations

### PR DESCRIPTION
**Pull request by nickdrozd**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: astroid/nodes/as_string.py [hunk 1]
```diff
@@ -50,15 +51,17 @@ def _docs_dedent(self, doc_node: Const | None) -> str:
             self.indent, doc_node.value.replace("\n", DOC_NEWLINE)
         )
 
-    def _stmt_list(self, stmts, indent=True):
+    def _stmt_list(self, stmts: list, indent: bool = True) -> str:
         """return a list of nodes to string"""
-        stmts = "\n".join(nstr for nstr in [n.accept(self) for n in stmts] if nstr)
-        if indent:
-            return self.indent + stmts.replace("\n", "\n" + self.indent)
+        stmts_str: str = "\n".join(
```
Comment (nickdrozd) [inline astroid/nodes/as_string.py:56]: Mypy doesn't like variables changed to different types.

Comment (Pierre-Sassoulas) [reply astroid/nodes/as_string.py:56]: Let's use two variable name instead. This is a pattern that is used a lot in old python but it's very error prone, imo it dying with mypy's adoption is a good thing).

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

File: astroid/bases.py [hunk 2]
```diff
@@ -620,16 +620,16 @@ def callable(self):
     def pytype(self) -> Literal["builtins.generator"]:
         return "builtins.generator"
 
-    def display_type(self):
+    def display_type(self) -> str:
```
Comment (DanielNoord) [inline astroid/bases.py:623]: We could have used `Literals` here. Not sure if it makes sense to change them now though...

Comment (Pierre-Sassoulas) [reply astroid/bases.py:623]: Yeah, we have a lot to do before starting to polish this to use litteral imo :D

Comment (DanielNoord) [reply astroid/bases.py:623]: For future PRs I do think we can start using them, if they make sense. Sometimes it's useful to know the specific string a function returns 😄

Comment (cdce8p) [reply astroid/bases.py:623]: > For future PRs I do think we can start using them, if they make sense. Sometimes it's useful to know the specific string a function returns 😄

I would disagree. We should only use `Literal` if there is a clear value to it. In most cases however it just adds unnecessary complexity. Example, what if the return string changes? You'll also have to update the `Literal` typing as well.

`Literal` is most useful for `@overloads` IMO.

Comment (DanielNoord) [reply astroid/bases.py:623]: I think for many of the methods that were being typed here the method could have also been a attribute, in which case I think it makes sense to do so as you would get similar information about the return value if it was an attribute.

But you are right, they shouldn't be used everywhere.

**Pull request merged**